### PR TITLE
Turned foreach loops into allocation free for loops

### PR DESCRIPTION
**Pull request by Amarcolina**
This PR turns all foreach loops in runtime code into for loops that do not allocate memory.  This should be a small help in our battle against performance!

### CODE REVIEW

File: Assets/LeapMotion/Scripts/DetectionUtilities/ExtendedFingerDetector.cs [hunk 1]
```diff
@@ -75,7 +75,8 @@ public class ExtendedFingerDetector : Detector {
     void OnValidate() {
       int required = 0, forbidden = 0;
       PointingState[] stateArray = { Thumb, Index, Middle, Ring, Pinky };
-      foreach(PointingState state in stateArray) {
```
Comment (jo3w4rd) [inline Assets/LeapMotion/Scripts/DetectionUtilities/ExtendedFingerDetector.cs]: OnValidate is only called in the editor, but okay...

Comment (Amarcolina) [inline Assets/LeapMotion/Scripts/DetectionUtilities/ExtendedFingerDetector.cs]: Ah yeah that's right, I didn't even notice it was in OnValidate.  Oh well :P

File: Assets/LeapMotion/Scripts/LeapHandController.cs [hunk 2]
```diff
@@ -98,7 +99,8 @@ public class LeapHandController : MonoBehaviour {
 
       /** Mark-and-sweep to finish unused HandRepresentations */
       HandRepresentation toBeDeleted = null;
-      foreach (KeyValuePair<int, HandRepresentation> r in all_hand_reps) {
+      for (var it = all_hand_reps.GetEnumerator(); it.MoveNext();) {
```
Comment (jselstad) [inline Assets/LeapMotion/Scripts/LeapHandController.cs]: This... this is good to know.

**Comment by jselstad** [comment]
Looks good!

**Pull request merged**